File: FE/src/types/TD_UngVien/TD_UngVien.ts
// DTO
export interface TD_UngVienDto {
  id: string;
  hoTen: string;
  ngaySinh: string; // yyyy-MM-dd
  gioiTinh?: number; // 0: Nam, 1: Nữ, 2: Khác
  email: string;
  soDienThoai: string;
  diaChi?: string;
  trinhDoHocVan?: string;
  kinhNghiem?: string;
  cvFile: string;
  ngayUngTuyen?: string; // yyyy-MM-dd
  thoiGianPhongVan?: string | null; // yyyy-MM-ddTHH:mm
  trangThai?: number;
  ghiChuUngVien?: string;
  tuyenDungId?: string;
  viTriTuyenDungText?: string;
  trangThaiText?: string;
  gioiTinhText?: string;
  createdDate?: string;
  updatedDate?: string;
}

export interface TD_UngVienTongQuanVM {
  totalCandidates: number;
  daNhanViec: number;
  dangChoPhongVan: number;
  daXetDuyet: number;
  daTuChoi: number;
  chuaXetDuyet: number;
  datPhongVan: number;
  interviewToday:number;
  interviewThisWeek: number;
  interviewThisMonth: number;
  avgInterviewPerDay: number;
  conversionRate: number;
  interviewRate: number;
}

// Search
export interface TD_UngVienSearch {
  HoTen?: string;
  Email?: string;
  sdt?: string;
  NgayUngTuyen?: string; // yyyy-MM-dd
  KinhNghiem?: string;
  GioiTinh?: number;
  TrangThai?: number;
  TuyenDungId?: string;
  pageIndex: number;
  pageSize: number;
  ThoiGianPhongVan_Start?: string; // yyyy-MM-dd
  ThoiGianPhongVan_End?: string;   // yyyy-MM-dd
}

// Create
export interface TD_UngVienCreate {
  HoTen: string;
  NgaySinh: string; // yyyy-MM-dd
  GioiTinh?: number;
  Email: string;
  SoDienThoai: string;
  DiaChi?: string;
  TrinhDoHocVan?: string;
  KinhNghiem?: string;
  CVFile: File; // upload
  NgayUngTuyen?: string; // yyyy-MM-dd
  ThoiGianPhongVan?: string; // yyyy-MM-ddTHH:mm
  TrangThai?: number;
  GhiChuUngVien?: string;
  TuyenDungId?: string;
}

// Edit
export interface TD_UngVienEdit extends TD_UngVienCreate {
  id: string;
}

// Create both UngVien and DonUngTuyen
export interface TD_UngVienAndDonUngTuyenCreate {
  // Thông tin ứng viên
  HoTen: string;
  NgaySinh?: string; // yyyy-MM-dd
  GioiTinh?: number;
  Email: string;
  SoDienThoai: string;
  DiaChi?: string;
  TrinhDoHocVan?: string;
  KinhNghiem?: string;
  CVFile?: File; // upload
  
  // Thông tin đơn ứng tuyển
  ViTriId: string;
  TrangThai?: number;
  GhiChu?: string;
  NgayNopDon: string; // yyyy-MM-dd
}

// Result DTO
export interface TD_UngVienAndDonUngTuyenResultDto {
  UngVien: TD_UngVienDto;
  DonUngTuyen: any; // TD_DonUngTuyenDto
  Message: string;
}

// Enum giới tính
export enum GioiTinhUngVien {
  Nam = 0,
  Nu = 1,
  Khac = 2,
} 

export enum TrangThaiUngVien {
  ChuaXetDuyet = 0,
  DaXetDuyet = 1,
  DangChoPhongVan = 2,
  DaNhanViec = 3,
  DaTuChoi = 4,
  DatPhongVan = 5
} 

File: BE/Hinet.Model/Entities/TuyenDung/TD_UngVien.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hinet.Model.Entities.TuyenDung
{
    [Table("TD_UngVien")]
    public class TD_UngVien : AuditableEntity
    {

        [StringLength(250)]
        public string? HoTen { get; set; }

        public DateTime? NgaySinh { get; set; }

        public GioiTinh_UngVien? GioiTinh { get; set; } // 0: Nam, 1: Nữ, 2: Khác

        [StringLength(250)]
        public string? Email { get; set; }

        [StringLength(250)]
        public string? SoDienThoai { get; set; }

        [StringLength(250)]
        public string? DiaChi { get; set; }

        [StringLength(200)]
        public string? TrinhDoHocVan { get; set; }

        [StringLength(500)]
        public string? KinhNghiem { get; set; }

        [StringLength(550)]
        public string CVFile { get; set; }

        public DateTime? NgayUngTuyen { get; set; }
        public DateTime? ThoiGianPhongVan { get; set; }
        public TrangThai_UngVien TrangThai { get; set; } = TrangThai_UngVien.ChuaXetDuyet;
        public string? GhiChuUngVien { get; set; }
        public Guid TuyenDungId { get; set; }
    }
    public enum TrangThai_UngVien
    {
        ChuaXetDuyet = 0, // Chưa xét duyệt
        DaXetDuyet = 1, // Đã xét duyệt
        DangChoPhongVan = 2, // Đang chờ phỏng vấn
        DaNhanViec = 3, // Đã nhận việc
        DaTuChoi = 4, // Đã từ chối
        DatPhongVan = 5 // Đạt Phỏng vấn
    }
    public enum GioiTinh_UngVien
    {
        Nam = 0,
        Nu = 1,
        Khac = 2
    }
}
